lib/cli-engine.js

```diff
@@ -506,6 +510,33 @@ CLIEngine.prototype = {
             ignore: ignoredPathsList
         };
 
+
+        /**
+         * Calculates the hash of the config file used to validate a given file
+         * @param  {string} filename The path of the file to retrieve a config object for to calculate the hash
+         * @returns {string}         the hash of the config
+         */
+        function hashOfConfigFor(filename) {
+            var config = configHelper.getConfig(filename);
```

Comment:
`getConfig` already returns the cached config of a file if you call it multiple times with the same filename, see https://github.com/eslint/eslint/blob/18a3f1bff2e9a36bb9672517ceb1453678fc6ca2/lib/config.js#L400-L404